ArchiSteamFarm/Commands.cs

```diff
@@ -1724,15 +1724,10 @@ internal sealed class Commands {
 								} else {
 									if (result.PurchaseResultDetail == EPurchaseResultDetail.CannotRedeemCodeFromClient) {
 										// If it's a wallet code, we try to redeem it first, then handle the inner result as our primary one
-										(EResult Result, EPurchaseResultDetail? PurchaseResult)? walletResult = await currentBot.ArchiWebHandler.RedeemWalletKey(key).ConfigureAwait(false);
-
-										if (walletResult != null) {
-											result.Result = walletResult.Value.Result;
-											result.PurchaseResultDetail = walletResult.Value.PurchaseResult.GetValueOrDefault(walletResult.Value.Result == EResult.OK ? EPurchaseResultDetail.NoDetail : EPurchaseResultDetail.BadActivationCode); // BadActivationCode is our smart guess in this case
-										} else {
-											result.Result = EResult.Timeout;
-											result.PurchaseResultDetail = EPurchaseResultDetail.Timeout;
-										}
+										(EResult Result, EPurchaseResultDetail PurchaseResult) walletResult = await currentBot.ArchiWebHandler.RedeemWalletKey(key).ConfigureAwait(false);
```

Comment:
You entirely removed the logic responsible for guessing proper purchase result detail if it's unavailable, this is also wrong.